File: app/src/main/java/han/ayeon/searchimgandvideo/view/savedlist/list/SavedListRecyclerViewAdapter.java
package han.ayeon.searchimgandvideo.view.savedlist.list;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import han.ayeon.searchimgandvideo.R;
import han.ayeon.searchimgandvideo.databinding.FragmentSavedItemBinding;;

public class SavedListRecyclerViewAdapter extends RecyclerView.Adapter<SavedListViewHolder> {

    private ArrayList<String> savedResultUrl;
    private FragmentSavedItemBinding dataBinding;


    @NonNull
    @Override
    public SavedListViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        dataBinding = DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()),
                R.layout.fragment_saved_item, parent, false);

        SavedListViewHolder savedListViewHolder = new SavedListViewHolder(dataBinding);

        return savedListViewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull SavedListViewHolder holder, int position) {

        if(savedResultUrl==null) return;
        String url = savedResultUrl.get(position);
        holder.showThumbNail(url);
    }

    @Override
    public int getItemCount() {
        if(savedResultUrl==null) return 0;
        else return savedResultUrl.size();
    }

    public void savedResultChange(ArrayList<String> savedResultUrl) {
        this.savedResultUrl = savedResultUrl;
        notifyDataSetChanged();
    }
}

File: app/src/main/java/han/ayeon/searchimgandvideo/presentation/view/savedlist/SavedListRecyclerViewAdapter.kt
package han.ayeon.searchimgandvideo.presentation.view.savedlist

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import han.ayeon.searchimgandvideo.R
import han.ayeon.searchimgandvideo.databinding.FragmentSavedItemBinding
import han.ayeon.searchimgandvideo.domain.data.Media

/**
 * Created by HanAYeon on 2019-07-18.
 */

class SavedListRecyclerViewAdapter : RecyclerView.Adapter<SavedListViewHolder>() {


    private var savedListItem = ArrayList<Media>()
    private lateinit var dataBinding : FragmentSavedItemBinding

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SavedListViewHolder {
        dataBinding = DataBindingUtil.inflate(LayoutInflater.from(parent.context),
                R.layout.fragment_saved_item, parent, false)

        return SavedListViewHolder(dataBinding)
    }


    override fun onBindViewHolder(holder: SavedListViewHolder, position: Int) {

        val url = savedListItem.get(position).thumbnailUrl
        holder.showThumbNail(url)

    }

    override fun getItemCount(): Int {
        return savedListItem.size
    }


    internal fun savedResultChange(resultData: java.util.ArrayList<Media>) {
        savedListItem = resultData
        notifyDataSetChanged()
    }


}